rpc/rest/security/rest_security.go

```diff
@@ -0,0 +1,304 @@
+			if err != nil {
+				a.log.Errorf("failed to hash password for user %s: %v", user.Name, err)
+				continue
+			}
+
+			a.userDb = append(a.userDb, &httpsecurity.User{
+				Name:        user.Name,
+				Password:    string(hash),
+				Permissions: user.Permissions,
+			})
+			a.log.Debugf("Registered user %s, permissions: %v", user.Name, user.Permissions)
+		}
+	}()
+
+	// Admin-group, available by default and always enabled for all URLs
+	a.AddPermissionGroup(&httpsecurity.PermissionGroup{
+		Name: admin,
+	})
+
+	a.registerSecurityHandlers()
+
+	return a
+}
+
+// AddPermissionGroup adds new permission group.
+func (a *authenticator) AddPermissionGroup(group ...*httpsecurity.PermissionGroup) {
+	for _, newPermissionGroup := range group {
+		for _, existingGroup := range a.groupDb {
+			if existingGroup.Name == newPermissionGroup.Name {
+				a.log.Warnf("permission group %s already exists, skipped")
+				continue
+			}
+		}
+		a.log.Debugf("added HTTP permission group %s", newPermissionGroup.Name)
+		a.groupDb = append(a.groupDb, newPermissionGroup)
+	}
+}
+
+// Validate the request
+func (a *authenticator) Validate(provider http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		authHeader := req.Header.Get(authHeaderStr)
+		if authHeader == "" {
+			a.formatter.Text(w, http.StatusUnauthorized, "401 Unauthorized: authorization header required")
+			return
+		}
+		bearerToken := strings.Split(authHeader, " ")
+		if len(bearerToken) != 2 {
+			a.formatter.Text(w, http.StatusUnauthorized, "401 Unauthorized: invalid authorization token")
+			return
+		}
+		token, err := jwt.Parse(bearerToken[1], func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, fmt.Errorf("error parsing token")
+			}
+			return []byte(secret), nil
+		})
+		if err != nil {
+			errStr := fmt.Sprintf("500 internal server error: %s", err)
+			a.formatter.Text(w, http.StatusInternalServerError, errStr)
+			return
+		}
+
+		if err := a.validateToken(token.Raw, req.URL.Path); err != nil {
+			errStr := fmt.Sprintf("401 Unauthorized: %v", err)
+			a.formatter.Text(w, http.StatusUnauthorized, errStr)
+			return
+		}
+
+		provider.ServeHTTP(w, req)
+	})
+}
+
+// Register authenticator-wide security handlers
+func (a *authenticator) registerSecurityHandlers() {
+	a.router.HandleFunc(login, a.createTokenEndpoint).Methods(http.MethodPost)
+	a.router.HandleFunc(logout, a.invalidateTokenEndpoint).Methods(http.MethodPost)
+}
+
+// Validates credentials and provides new token
+func (a *authenticator) createTokenEndpoint(w http.ResponseWriter, req *http.Request) {
+	name, errCode, err := a.validateCredentials(req)
+	if err != nil {
+		a.formatter.Text(w, errCode, err.Error())
+		return
+	}
+	var claims jwt.MapClaims = make(map[string]interface{})
+	if a.expTime != 0 {
+		claims["exp"] = time.Now().Add(time.Minute * time.Duration(a.expTime)).Unix()
```

Comment:
You should probably use `ExpiresAt` field from [StandardClaims](https://godoc.org/github.com/dgrijalva/jwt-go#example-NewWithClaims--StandardClaims) for this. 

Also, you should store username here as well.